torch/distributed/nn/functional.py

```diff
@@ -94,20 +116,23 @@ def all_gather(tensor, group=dist.group.WORLD):
     return _AllGather.apply(group, tensor)
 
 
-def all_to_all(tensors, group=dist.group.WORLD):
+def all_to_all(tensors, group=dist.group.WORLD, out_tensor_list=None):
```

Comment:
We should make the signature of this all_to_all consistent with https://github.com/pytorch/pytorch/blob/master/torch/distributed/distributed_c10d.py#L2483 and use input_split_sizes and output_split_sizes.

Ideally in nn.Linear code we should just have to replace dist.all_to_all single with the functional version and not have to change any other lines.